trace-sdk/src/test/java/io/bitrise/trace/network/MetricSenderIntegrationTest.java

```diff
@@ -0,0 +1,77 @@
+package io.bitrise.trace.network;
+
+import androidx.annotation.NonNull;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.bitrise.trace.configuration.ConfigurationManager;
+import io.bitrise.trace.test.DataTestUtils;
+import io.bitrise.trace.test.MetricTestProvider;
+import io.opencensus.proto.metrics.v1.Metric;
+import retrofit2.Response;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Integration tests that send a metric to the backend and expect confirmation by headers.
+ */
+public class MetricSenderIntegrationTest {
```

Comment:
These tests are validating the same dummy values against the backend, but I think the ticket was to collect real data. Please update them accordingly.

Note: due to their nature (firing up real network calls), we should annotate them with LargeTest annotation to indicate it takes more time to complete.

Reply:
Good point. I had to move these tests then to `androidTest` not `test`.

I have updated all the tests except the ApplicationStartUp to call the collector, then the formatter and send the formatted data across. 

There was an issue with the ApplicationStartUp in that, the sdk doesn't actually have any screens - I could have mocked all the calls but I think that might be over complicating things. Instead, I checked what data got created when it called `onDataCollected` - this then still gets passed through the formatter. Does that sound okay?

Reply:
Sure, thanks for the updates!